src/config/configuration.ts

```diff
@@ -1,41 +1,45 @@
-import {ITransportStrategy} from './transport/transport-strategy';
+import {TransportStrategy} from './transport/transport-strategy';
 
-export interface IConfiguration {
+export interface SimpleCacheConfiguration {
   // TODO: add RetryStrategy
   // TODO: add Middlewares
-  getTransportStrategy(): ITransportStrategy;
-  withTransportStrategy(transportStrategy: ITransportStrategy): IConfiguration;
+  getTransportStrategy(): TransportStrategy;
+  withTransportStrategy(
+    transportStrategy: TransportStrategy
+  ): SimpleCacheConfiguration;
   getMaxIdleMillis(): number;
-  withMaxIdleMillis(maxIdleMillis: number): IConfiguration;
-  withClientTimeout(clientTimeout: number): IConfiguration;
+  withMaxIdleMillis(maxIdleMillis: number): SimpleCacheConfiguration;
+  withClientTimeout(clientTimeout: number): SimpleCacheConfiguration;
 }
 
-export class Configuration implements IConfiguration {
-  private readonly transportStrategy: ITransportStrategy;
+export class Configuration implements SimpleCacheConfiguration {
```

Comment:
naming nitpick (not a blocker):
`Configuration implements IConfiguration` rolls of the tongue easier than `Configuration implements SimpleCacheConfiguration` I think. 

Should this be  `CacheConfiguration implements SimpleCacheConfiguration`?

Reply:
If we go with that, should Laptop be LaptopCacheConfiguration?